crates/bevy_ecs/src/resource/resource_query.rs

```diff
@@ -143,7 +184,7 @@ pub trait FetchResource<'a>: Sized {
     fn release(resources: &Resources);
 
     #[allow(clippy::missing_safety_doc)]
-    unsafe fn get(resources: &'a Resources, system_id: Option<SystemId>) -> Self::Item;
+    unsafe fn get(resources: &'a Resources, system_id: Option<SystemId>) -> (Self::Item, bool);
```

Comment:
Lets break this up into `get()` and `is_some()`. I think it will make the code easier to read and will allow us to avoid retrieving the actual data for queries that dont "match".

Reply:
Yes it is really nicer this way